reactioncommerce/reaction-next-starterkit
# feat: breadcrumb component of product grid (tags) and PDP

Resolves #78 

## Issue

> Create Breadcrumb Navigation component for use on Product Grid and PDP.
> 
> The breadcrumb navigation should start with a home link which should link back to the index.
> 
> After that it should drill down into the tag category that the user is currently browsing, including the parents in the breadcrumb list if applicable.
> e.g. if browsing the 'Skirts' tag within Womens and Active Wear, the breadcrumb list should render:
> Home > Womens > Active Wear > Skirts
> 
> If on a PDP page, the breadcrumbs should show the breadcrumb list where the user originated followed by the product title

## Solution

Create a Breadcrumb component that displays a hierarchal path of crumbs on a `tag` page (i.e. `Home > {parentTag} > {childtag}` when on a child tag page), or a navigational path of crumbs on a PDP (i.e. `Home > {parentTag} > {childtag} > product` when coming from a tag page, or (i.e. `Home > {product}` when coming from the default grid.

A new HOC, `withTags` was created and is available at the `_app` level. This HOC provides all tags available on the shop.

## Testing
- Create multiple products, with multiple tags attached to each. Tags should be both top level tags, and child tags of these top level tags
- Navigate to a top level tag page, and see _clickable_ breadcrumbs: `Home > {tag}`
- Navigate back to the home page, then to a child tag page, and see _clickable_ breadcrumbs: `Home > {tag} > {childTag}` - you should see the parent tag even though you navigated directly to a child tag
- Navigate to a product from the childTag page - you should see the navigation history of how you got to this page as the breadcrumbs: `Home > {tag} > {childTag} > {productName}`
- Navigate back to the home page, and then to any product. You should see `Home > {productName}`
